src/NuGet.Core/NuGet.Build.Tasks/RestoreTaskEx.cs

```diff
@@ -288,14 +290,22 @@ internal string GetProcessFileName(string processFileName)
                 }
             }
 #endif
-            msBuildGlobalProperties["ExcludeRestorePackageImports"] = "true";
+            if (globalProperties != null)
+            {
+                foreach (KeyValuePair<string, string> item in globalProperties)
+                {
+                    yield return item;
+                }
+            }
+
+            yield return new KeyValuePair<string, string>("ExcludeRestorePackageImports", bool.TrueString);
```

Comment:
Do you always want to return this property? Will there be cases in which `IBuildEngine6.GetGlobalProperties()` (.NET core) also contains `ExcludeRestorePackageImports` property? Would it be good to add a check to avoid returning `ExcludeRestorePackageImports` twice?

This also applies for `OriginalMSBuildStartupDirectory`.

Reply:
I think its okay in this case if there are duplicates because the other process that reads the properties will de-dupe them where the last one wins.  In the case of `ExcludeRestorePackageImports` and `OriginalMSBuildStartupDirectory`, if they are set, we would still want to override the values (which has been the behavior before).  That said, I don't think these would ever be set by anyone.